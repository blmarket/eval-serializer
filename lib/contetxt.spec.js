"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const context_1 = require("./context");
const test_model_1 = require("./test-model");
describe("ES module", () => {
    const ctx = new context_1.default();
    ctx.add(test_model_1.Row, r => [r.label, r.children]);
    ctx.add(test_model_1.PlainLink, x => [x.label, x.url, x.children]);
    ctx.add(test_model_1.Link, x => [x.label, x.url, x.color, x.children]);
    ctx.add(test_model_1.WikiLink, x => [x.label, x.url, x.children]);
    const sample = new test_model_1.Row("row:label", [
        new test_model_1.WikiLink("wiki", "wikiurl"),
        new test_model_1.PlainLink("link", "linkurl"),
        new test_model_1.Link("link", "linkurl", "#dddddd"),
    ]);
    it("Serializes sample object well enough", () => {
        const str = ctx.serialize(sample);
        const inst2 = new Function('Row', 'PlainLink', 'Link', 'WikiLink', 'return ' + str)(test_model_1.Row, test_model_1.PlainLink, test_model_1.Link, test_model_1.WikiLink);
        expect(str).toMatchSnapshot();
        expect(ctx.serialize(inst2)).toEqual(str);
    });
    it("Deserializes string", () => {
        const str = 'new WikiLink("wiki", "wikiurl", [])';
        const tmp = ctx.deserialize(str);
        expect(tmp).toMatchSnapshot();
    });
});
