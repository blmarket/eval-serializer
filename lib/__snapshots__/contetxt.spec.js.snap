// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ES module Deserializes string 1`] = `
WikiLink {
  "children": Array [],
  "color": "#e3f2fd",
  "label": "wiki",
  "type": "Link",
  "url": "wikiurl",
}
`;

exports[`ES module Serializes sample object well enough 1`] = `
"new Row(\\"row:label\\", [
  new WikiLink(\\"wiki\\", \\"wikiurl\\", []),
  new PlainLink(\\"link\\", \\"linkurl\\", []),
  new Link(\\"link\\", \\"linkurl\\", \\"#dddddd\\", []) ])"
`;
